rasa/nlu/components.py

```diff
@@ -649,17 +655,43 @@ def can_handle_language(cls, language: Hashable) -> bool:
             `True` if component can handle specific language, `False` otherwise.
         """
 
-        # if language_list is set to `None` it means: support all languages
+        # If both `supported_language_list` and `not_supported_language_list` are set to `None`,
+        # it means: support all languages
         if language is None or (
             cls.supported_language_list is None
             and cls.not_supported_language_list is None
         ):
             return True
 
-        language_list = cls.supported_language_list or []
-        not_supported_language_list = cls.not_supported_language_list or []
+        # check language supporting settings
+        if cls.supported_language_list and cls.not_supported_language_list:
+            # When user set both language supporting settings to not None, it will lead to ambiguity.
+            raise Exception(
+                "Only one of `supported_language_list` and `not_supported_language_list` can be set to not None"
+            )
 
-        return language in language_list or language not in not_supported_language_list
+        # convert to `list` for membership test
+        supported_language_list = (
+            cls.supported_language_list
+            if cls.supported_language_list is not None
+            else []
+        )
+        not_supported_language_list = (
+            cls.not_supported_language_list
+            if cls.not_supported_language_list is not None
+            else []
+        )
+
+        # check if user provided a available setting
+        if not supported_language_list and not not_supported_language_list:
+            # One of language settings must be available (not None and not a empty list),
+            # There are three combinations of settings are not available: (None, []), ([], None) and ([], [])
+            raise Exception("Empty language list is not a available setting")
```

Comment:
```suggestion
            raise Exception("Empty lists for both `supported_language_list` and `not_supported language_list` is not a valid setting. If you meant to allow all languages for the component use `None` for both of them.")
```
Please change to `RasaException` before accepting the change.

Reply:
suggestion accepted